include/range/v3/detail/config.hpp

```diff
@@ -22,15 +22,48 @@
 #define RANGES_ASSERT assert
 #endif
 
+#ifndef __has_builtin
+#define __has_builtin(x) 0
```

Comment:
Sorry, I meant to comment here. Who says `__has_builtin` is necessarily a macro, even on platforms that support it? And `#define __has_builtin` is technically not portable. Is there a better way?